Mage.Sets/src/mage/sets/conspiracytakethecrown/LeovoldEmissaryOfTrest.java

```diff
@@ -0,0 +1,194 @@
+ *        provided with the distribution.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY BetaSteward_at_googlemail.com ``AS IS'' AND ANY EXPRESS OR IMPLIED
+ *  WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ *  FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL BetaSteward_at_googlemail.com OR
+ *  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ *  SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ *  ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ *  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+ *  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *  The views and conclusions contained in the software and documentation are those of the
+ *  authors and should not be interpreted as representing official policies, either expressed
+ *  or implied, of BetaSteward_at_googlemail.com.
+ */
+package mage.sets.conspiracytakethecrown;
+
+import java.util.HashSet;
+import java.util.UUID;
+import mage.MageInt;
+import mage.abilities.Ability;
+import mage.abilities.TriggeredAbilityImpl;
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.effects.ContinuousRuleModifyingEffectImpl;
+import mage.abilities.effects.common.DrawCardSourceControllerEffect;
+import mage.cards.CardImpl;
+import mage.constants.*;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.permanent.Permanent;
+import mage.players.Player;
+import mage.watchers.Watcher;
+
+/**
+ *
+ * @author maxlebedev
+ */
+public class LeovoldEmissaryOfTrest extends CardImpl {
+
+    public LeovoldEmissaryOfTrest(UUID ownerId) {
+        super(ownerId, 77, "Leovold, Emissary of Trest", Rarity.MYTHIC, new CardType[]{CardType.CREATURE}, "{B}{G}{U}");
+        this.expansionSetCode = "CN2";
+        this.supertype.add("Legendary");
+        this.subtype.add("Elf");
+        this.subtype.add("Advisor");
+        this.power = new MageInt(3);
+        this.toughness = new MageInt(3);
+
+        // Each opponent can't draw more than one card each turn.  (Based on SpiritOfTheLabyrinth)
+        this.addAbility(new SimpleStaticAbility(Zone.BATTLEFIELD, new LeovoldEmissaryOfTrestEffect()), new LeovoldEmissaryOfTrestWatcher());
+
+        // Whenever you or a permanent you control becomes the target of a spell or ability an opponent controls, you may draw a card.
+        this.addAbility(new LeovoldEmissaryOfTrestTriggeredAbility());
+    }
+
+    public LeovoldEmissaryOfTrest(final LeovoldEmissaryOfTrest card) {
+        super(card);
+    }
+
+    @Override
+    public LeovoldEmissaryOfTrest copy() {
+        return new LeovoldEmissaryOfTrest(this);
+    }
+}
+
+
+class LeovoldEmissaryOfTrestWatcher extends Watcher {
+
+    private final HashSet<UUID> opponentsThatDrewCard;
```

Comment:
This won't work correctly, if the controller of Leovold, Emissary of Trest changes.
The watcher should watch and save for all players if a card was drew.
The check if an player is an opponent should happen for the rule modifying effect.